zerver/lib/bugdown/__init__.py

```diff
@@ -545,6 +545,18 @@ def youtube_image(self, url: Text) -> Optional[Text]:
             return "https://i.ytimg.com/vi/%s/default.jpg" % (yt_id,)
         return None
 
+    def get_youtube_title(self, url: Text) -> Optional[Text]:
+        request_url = "https://noembed.com/embed?url=" + url
+        request_url = sanitize_url(request_url)
+        try:
+            response = requests.get(request_url).json()
+            youtube_title = response['title']
+            return youtube_title
+        except ConnectionError:
```

Comment:
I think checking for `Exception` is better than checking for `ConnectionError`. It's possible that you may get many other exceptions here as well.